### PR TITLE
support searching customized string in the search box

### PR DESCRIPTION
**Pull request by Lijiaoa**
### Description ###
- PR title, support search look like `10`(trial No.) and `nj`(trial id) string in search input
- add filter trial `Default metric` in filter `button`

### CODE REVIEW

File: ts/webui/src/components/experiment/trialdetail/table/tableFunction/search/Search.tsx [hunk 1]
```diff
@@ -142,102 +171,124 @@ function Search(props): any {
             alert('Please delete special characters in the conditions!');
             return;
         }
-        // according [input val] to change searchFilter list
-        const allFilterConditions = searchInputVal.trim().split(';');
         const newSearchFilter: any = [];
+        const str = searchInputVal.trim();
+        // user input: string and this string don't include [: < > ≠]
+        if (str.includes('>') || str.includes('<') || str.includes(':') || str.includes('≠')) {
```
Comment (liuzhe-lz) [inline ts/webui/src/components/experiment/trialdetail/table/tableFunction/search/Search.tsx:177]: We are currently using "≠" character for non-equal filter?

File: ts/webui/src/components/experiment/trialdetail/table/tableFunction/search/Search.tsx [hunk 2]
```diff
@@ -142,102 +171,124 @@ function Search(props): any {
...
+            // according [input val] to change searchFilter list
+            const allFilterConditions = searchInputVal.trim().split(';');
+            // delete '' in filter list
+            if (allFilterConditions.includes('')) {
```
Comment (liuzhe-lz) [inline ts/webui/src/components/experiment/trialdetail/table/tableFunction/search/Search.tsx:181]: This only removes first empty string.
It should be `allFilterConditions.filter(item => item)`

File: ts/webui/src/components/experiment/trialdetail/table/tableFunction/search/searchFunction.ts [hunk 3]
```diff
@@ -90,6 +90,20 @@ const getTrialsBySearchFilters = (
             arr = arr.filter(trial => trial.sequenceId.toString() === element.value1);
         } else if (element.name === 'StatusNNI') {
             arr = searchChoiceFilter(arr, element, 'status');
+        } else if (element.name === 'Default metric') {
+            if (element.operator === '=') {
+                arr = arr.filter(trial => trial.latestAccuracy! === JSON.parse(element.value1));
```
Comment (matluster) [inline ts/webui/src/components/experiment/trialdetail/table/tableFunction/search/searchFunction.ts:95]: What if trial doesn't have a `trial.latestAccuracy`?

Comment (Lijiaoa) [reply ts/webui/src/components/experiment/trialdetail/table/tableFunction/search/searchFunction.ts:95]: 页面正常，没有 latestAccuracy 的 trial 不会被筛选出来。PR完整后会再测下

File: ts/webui/src/components/experiment/trialdetail/table/tableFunction/search/Search.tsx (hunk 1 above)
Comment (Lijiaoa) [reply ts/webui/src/components/experiment/trialdetail/table/tableFunction/search/Search.tsx:177]: by origin design, we use `≠` to show these status `!=`. but seems `!=` is better. and I seems `!==` is more smooth

Comment (Lijiaoa) [reply ts/webui/src/components/experiment/trialdetail/table/tableFunction/search/Search.tsx:177]: image

**Review by ultmaster — APPROVED** [review]

File: ts/webui/src/components/experiment/trialdetail/table/tableFunction/search/Search.tsx (hunk 2 above)
Comment (Lijiaoa) [reply ts/webui/src/components/experiment/trialdetail/table/tableFunction/search/Search.tsx]: updated

**Comment by Lijiaoa** [comment]
the code for `search input` will refactor in someday, because current implement logic looks complicated, although it works well

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**